packages/react-router/lib/router/router.ts

```diff
@@ -770,6 +770,10 @@ const defaultMapRouteProperties: MapRoutePropertiesFunction = (route) => ({
 
 const TRANSITIONS_STORAGE_KEY = "remix-router-transitions";
 
+// Flag used on new `loaderData` to indicate that we do not want to preserve
+// any prior loader data from the throwing route in `mergeLoaderData`
+const ResetLoaderDataSymbol = Symbol("ResetLoaderData");
```

Comment:
We used to use `undefined` to detect this scenario, switched to a symbol since `undefined` is a valid value now